plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -418,6 +422,49 @@ const updateBusinessLocation = ( countryAndState: string ) => {
 	} );
 };
 
+const updateStoreCurrency = async ( countryAndState: string ) => {
+	const { general: settings = {} } = await resolveSelect(
+		SETTINGS_STORE_NAME
+	).getSettings( 'general' );
+
+	const countryCode = getCountryCode( countryAndState ) as string;
+	const { currencySymbols = {}, localeInfo = {} } = getAdminSetting(
+		'onboarding',
+		{}
+	);
+	const currencySettings = CurrencyFactory().getDataForCountry(
+		countryCode,
+		localeInfo,
+		currencySymbols
+	) as {
```

Comment:
small nit - is it possible to do this without using `as` ? seems like `getDataForCountry()` should have the correct types

Reply:
Good catch!

`currencySettings` works without `as`, but the dispatch complains about typing without `as` 😞 

So it's either we type `currencySettings` or add `as string`. I don't have a strong preference. Which one do you prefer?
```
{
	woocommerce_currency: currencySettings.code as string,
	woocommerce_currency_pos: currencySettings.symbolPosition as string,
	woocommerce_price_thousand_sep: currencySettings.thousandSeparator as string,	
}
```

![Screen Shot 2023-09-13 at 10 41 31 PM](https://github.com/woocommerce/woocommerce/assets/4723145/60c56179-9ebc-4a1f-a465-025b6f4e8dd3)

Reply:
A stricter check like this should satisfy TS 😀 
There's still an outstanding error there with the symbol because our version of TS doesn't correctly understand [].includes as type narrowing, so might have to expand that out so that TS gets it

<img width="677" alt="image" src="https://github.com/woocommerce/woocommerce/assets/27843274/f92cd110-3c3d-4c5c-abbf-1785bdeba3da">

Reply:
Let's come back to this after adding zod. I think `as` approach is simpler.